protobuf/v1/snapshot.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+
+import "common.proto";
+package mayastor.v1;
+
+service SnapshotRpc {
+    // Common snapshot related methods.
+    rpc ListSnapshot(ListSnapshotsRequest) returns (ListSnapshotsResponse) {}
+    rpc DeleteSnapshot(DeleteSnapshotRequest) returns (google.protobuf.Empty) {}
+}
+/// Request parameters to list snapshot.
+message ListSnapshotsRequest {
+    optional string source_uuid = 1;  // source uuid from which snapshot is taken
+}
+
+/// List Snapshot Response.
+message ListSnapshotsResponse {
+    repeated ReplicaSnapshot snapshots = 1; // list of Snapshot Params.
+}
+/// Delete Snapshot Request.
+message DeleteSnapshotRequest {
+    string snapshot_uuid = 1; // snapshot to be deleted
```

Comment:
This is not enough to ensure idempotency: GTM-913
Why is create snapshot not here?

Reply:
> This is not enough to ensure idempotency: GTM-913 Why is create snapshot not here?

1. Currently We are exposing 2 gRPC interface for Create_snapshot. One for replica and another for nexus. That is the reason, create_snapshot gRPC I kept it as part of respective proto buff (replica and nexus). But We can discuss if We can move both the gRPC request to the snapshot gRPC file.
2. Some changes are needed in parameter list as part of GTM-913. Currently those new parameters addition is not consider as part of this PR.

Reply:
I think we should move it here, otherwise seems a big clunky to have a snapshot service just for some calls and not others..

For 2 I think we should consider this as part of the PR because it's an important nuance here, if we can't have a completely generic snapshot delete, we need the replica or pool id in order to ensure that the snapshot is deleted. In such case if the delete also moves away from this common interface, we end up with a common interface that just does list? And even the list is just for replica snapshots?

Reply:
Handled both the scenarios. Please check once.